pkg/networkoptions/network_options.go

```diff
@@ -62,6 +62,31 @@ func networkOptionFromString(s string) NetworkOption {
 	return Undefined
 }
 
+func (n NetworkOption) ModelNetwork(devnetEndpoint string) (models.Network, error) {
```

Comment:
we should not name this as devnet. I would suggest naming it custom as currently we name it devnet but in the future we might have other networks to support.